### ISSUE TITLE
Under some exit circumstances DEPNotify or jamfHelper may not be ended

### ISSUE THREAD

**Issue by ryangball** [grahampugh/erase-install#146]
**Describe the bug**
Under some exit circumstances DEPNotify or jamfHelper may not be ended as they are being killed in specific areas of the script but the script might exit abruptly causing DEPNotify or jamfHelper to stay open (as well as other cleanup processes potentially not occurring).

**To Reproduce**
 I have seen this running the script both from Jamf Self Service as well as locally. Again, this is a termination of the script execution at times when you would not expect it. Most easily reproduced by running from terminal and using ctrl+c to cancel the process (which again, is abrupt and unexpected but edge cases are a thing). 
 
**Expected behavior**
When the script exits at any point, the things we need to clean up or terminate should be done if at all possible.

I suggest using a `trap` on EXIT. Using a `trap` will ensure that anything we need to end or clean up will be done upon exiting. This will also clean up duplicate multiple kill_process functions to kill the same thing but done at different points in the script. Consider the following example:

```
#!/bin/bash

breadcrumb="/private/tmp/breadcrumb.txt"

finish() {
    [[ -e "$breadcrumb" ]] && /bin/rm "$breadcrumb"
    echo "Done"
}

trap "finish" EXIT

touch "$breadcrumb"

echo "Press ctrl+c any time to exit and clean up"

for num in {1..30}; do
    echo "$num"
    sleep 1
done

/bin/rm "$breadcrumb"

exit 0
```

FYI: When a have a little more time I'll create a PR. As always, great work with this @grahampugh.

**Comment by grahampugh** [comment]
Good idea. I was not aware of `trap` as an option.

**Comment by grahampugh** [comment]
If you have a PR in the works, please submit it to the v25.0-rc branch. Thanks :)

**Comment by grahampugh** [comment]
Should be fixed now in v25.0. Let me know if not.
